File: java/src/main/java/io/amerhwitat/bizx/game/AAASystems.java
package io.amerhwitat.bizx.game;

import java.util.*;

/** Engine-neutral systemic simulation: world cells, NPC needs, events, economy and abilities. */
public final class AAASystems {
    public record WorldCell(String id,long seed,String biome,int population,String weather,boolean loaded) {}
    public record NPC(String id,String faction,double food,double rest,double safety,String state) {}
    public record Vehicle(String id,String owner,double fuel,double durability,int cargoCapacity,int cargo,double speed) {}
    private final Random rng; private long tick;
    private final Map<String,WorldCell> cells=new HashMap<>(); private final Map<String,NPC> npcs=new HashMap<>();
    private final Map<String,Double> prices=new HashMap<>(); private final List<Map<String,Object>> events=new ArrayList<>();
    private final Map<String,Map<String,Object>> abilities=new HashMap<>();
    public AAASystems(long seed){rng=new Random(seed);}
    public void addCell(WorldCell c){cells.put(c.id(),c);} public void addNpc(NPC n){npcs.put(n.id(),n);}
    public void registerAbility(String id,double resource,int cooldown,Map<String,Object> effects){abilities.put(id,Map.of("resource",resource,"cooldown",cooldown,"effects",effects));}
    public void triggerEvent(String id,String kind,int duration,Map<String,Object> effects){events.add(new HashMap<>(Map.of("id",id,"kind",kind,"remaining",duration,"effects",effects)));}
    public void tick(){tick++; npcs.replaceAll((id,n)->new NPC(n.id(),n.faction(),Math.max(0,n.food()-.005),Math.max(0,n.rest()-.003),n.safety(),n.safety()<.25?"flee":n.food()<.25?"seek_food":n.rest()<.2?"sleep":"work")); events.forEach(e->e.put("remaining",((int)e.get("remaining"))-1)); events.removeIf(e->(int)e.get("remaining")<=0);}
    public double market(String good,double supply,double demand){double p=Math.max(.01,(demand+1)/(supply+1));prices.put(good,p);return p;}
    public Map<String,Object> resolveAbility(String id,int baseDamage,int defense){int roll=rng.nextInt(20)+1;boolean hit=roll+5>=defense;return Map.of("hit",hit,"critical",roll==20,"damage",hit?baseDamage*(roll==20?2:1):0,"effects",abilities.getOrDefault(id,Map.of()).getOrDefault("effects",Map.of()));}
    public long tickCount(){return tick;} public Map<String,Double> prices(){return Map.copyOf(prices);}
}

File: kotlin/src/main/kotlin/io/amerhwitat/bizx/game/AAASystems.kt
package io.amerhwitat.bizx.game

data class WorldCell(val id:String,val seed:Long,val biome:String,var population:Int=0,var weather:String="clear",var loaded:Boolean=false)
data class Npc(val id:String,val faction:String,var food:Double=1.0,var rest:Double=1.0,var safety:Double=1.0,var state:String="idle")
data class Vehicle(val id:String,val owner:String,var fuel:Double=100.0,var durability:Double=100.0,var cargo:Int=0,val cargoCapacity:Int=10,var speed:Double=1.0)

class AAASystems(seed:Long=1L) {
    private var state=seed; var tick:Long=0; val cells=mutableMapOf<String,WorldCell>(); val npcs=mutableMapOf<String,Npc>(); val prices=mutableMapOf<String,Double>(); val events=mutableListOf<MutableMap<String,Any>>(); val abilities=mutableMapOf<String,Map<String,Any>>()
    private fun rand():Int { state = (state*6364136223846793005L + 1442695040888963407L); return ((state ushr 33) and Int.MAX_VALUE.toLong()).toInt()%20+1 }
    fun addCell(c:WorldCell)=cells.set(c.id,c); fun addNpc(n:Npc)=npcs.set(n.id,n)
    fun registerAbility(id:String,resource:Double,cooldown:Int,effects:Map<String,Any>) { abilities[id]=mapOf("resource" to resource,"cooldown" to cooldown,"effects" to effects) }
    fun triggerEvent(id:String,kind:String,duration:Int,effects:Map<String,Any>) { events += mutableMapOf("id" to id,"kind" to kind,"remaining" to duration,"effects" to effects) }
    fun tickOnce(){ tick++; npcs.values.forEach{it.food=(it.food-.005).coerceAtLeast(0.0);it.rest=(it.rest-.003).coerceAtLeast(0.0);it.state=when{it.safety<.25->"flee";it.food<.25->"seek_food";it.rest<.2->"sleep";else->"work"}};events.forEach{it["remaining"]=(it["remaining"] as Int)-1};events.removeAll{it["remaining"] as Int<=0} }
    fun market(good:String,supply:Double,demand:Double):Double { val p=maxOf(.01,(demand+1)/(supply+1));prices[good]=p;return p }
    fun resolveAbility(id:String,baseDamage:Int,defense:Int=0):Map<String,Any>{val r=rand();val hit=r+5>=defense;return mapOf("hit" to hit,"critical" to (r==20),"damage" to if(hit) baseDamage*if(r==20)2 else 1 else 0,"effects" to (abilities[id]?.get("effects") ?: emptyMap<String,Any>()))}
}
